libnavui-voice/api/current.txt

```diff
@@ -101,12 +101,11 @@ package com.mapbox.navigation.ui.voice.view {
     ctor public MapboxSoundButton(android.content.Context context, android.util.AttributeSet? attrs);
     ctor public MapboxSoundButton(android.content.Context context, android.util.AttributeSet? attrs, int defStyleAttr);
     method public void mute(com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean>? callback);
-    method public void muteAndExtend(long duration, com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean> callback);
+    method public void muteAndExtend(long duration, com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean>? callback);
     method public void unmute(com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean>? callback);
-    method public void unmuteAndExtend(long duration, com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean> callback);
+    method public void unmuteAndExtend(long duration, com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer<java.lang.Boolean>? callback);
```

Comment:
`muteAndExtend` and `unmuteAndExtend` are changed signature and allow _nullable_ _callback_ to made them consistent with `unmute` and `mute`

cc @abhishek1508 @Guardiola31337

Reply:
The reason `muteAndExtend` and `unmuteAndExtend` has non nullable callback because there is an animation that is happening. If you don't pass a callback your `isMuted` variable in the example activity will immediately be changed to the state and you would be able to call the API again. This can cause issues.
I hope it makes sense @RingerJK 

I don't think we should make these callbacks nullable in the `extend` functions.

Reply:
@abhishek1508 yeah, totally make sense for the previous approach when we blocked interaction with the button when the animation was running. 
It's a bit outdated as we changed flow, now mute/unmute changed immediately and a user is not waiting for animation and can interact with SoundButton immediately. You could try to run the current branch and click multiple time on SoundButton: a state is changed immediately, without "pause" in the animation

Reply:
Should it be defaulted to `null` (`@JvmOverloads` for Java users) to avoid forcing users to give in a callback?

Reply:
`null` is also a value, which means you don't need a result. if we add `@ JvmOverloads`, there'll be 3 possible values to pass: _nothing_, `null`, and `callback`. _nothing_ and `null` are the same, which is wired that both possible. 
In a way, as you want to go the best variant is add additional methods without params and change currents to non-null params. 
But I'd keep API as simple as possible and don't add additional methods which don't bring a value

Reply:
IMO it brings the value of not needing to pass the value explicitly if you don't need the callback ¯\\\_(ツ)\_/¯